tests/models/test_metatensor.py

```diff
@@ -0,0 +1,103 @@
+import pytest
+import torch
+from ase.build import bulk
+
+from torch_sim.io import atoms_to_state
+from torch_sim.models.interface import validate_model_outputs
+from torch_sim.state import SimState
+
+
+try:
+    from metatensor.torch.atomistic.ase_calculator import MetatensorCalculator
+    from metatrain.utils.io import load_model
+
+    from torch_sim.models.metatensor import MetatensorModel
+except ImportError:
+    pytest.skip("Metatensor not installed", allow_module_level=True)
+
+
+@pytest.fixture
+def dtype() -> torch.dtype:
+    """Fixture to provide the default dtype for testing."""
+    return torch.float32
+
+
+@pytest.fixture
+def metatensor_calculator(device: torch.device):
+    """Load a pretrained metatensor model for testing."""
+    return MetatensorCalculator(
+        model=load_model(
+            "https://huggingface.co/lab-cosmo/pet-mad/resolve/main/models/pet-mad-latest.ckpt"
+        ).export(),
+        device=device,
+    )
+
+
+@pytest.fixture
+def cu_system(dtype: torch.dtype, device: torch.device) -> SimState:
+    # Create FCC Copper
+    cu_fcc = bulk("Cu", "fcc", a=3.58, cubic=True)
+    return atoms_to_state([cu_fcc], device, dtype)
+
+
+@pytest.fixture
+def metatensor_model(device: torch.device) -> MetatensorModel:
+    """Create an MetatensorModel wrapper for the pretrained model."""
+    return MetatensorModel(
+        model="pet-mad",
+        device=device,
+    )
+
+
+def test_metatensor_initialization(device: torch.device) -> None:
+    """Test that the metatensor model initializes correctly."""
+    model = MetatensorModel(
+        model="pet-mad",
+        device=device,
+    )
+    assert model._device == device  # noqa: SLF001
+    assert model._dtype == torch.float32  # noqa: SLF001
+
+
+def test_metatensor_calculator_consistency(
+    metatensor_model: MetatensorModel,
+    metatensor_calculator: MetatensorCalculator,
+    cu_system: SimState,
+    device: torch.device,
+) -> None:
+    """Test consistency between MetatensorModel and MetatensorCalculator."""
+    # Get MetatensorModel results
+    metatensor_results = metatensor_model.forward(cu_system)
+
+    # Set up ASE calculator
+    cu_fcc = bulk("Cu", "fcc", a=3.58, cubic=True)
+    cu_fcc.calc = metatensor_calculator
+
+    # Get calculator results
+    calc_energy = cu_fcc.get_potential_energy()
+    calc_forces = torch.tensor(
+        cu_fcc.get_forces(),
+        device=device,
+        dtype=metatensor_results["forces"].dtype,
+    )
+
+    # Test consistency with reasonable tolerances
+    torch.testing.assert_close(
+        metatensor_results["energy"].item(),
+        calc_energy,
+        atol=1e-4,
+        rtol=0.5,
+    )
+    torch.testing.assert_close(
+        metatensor_results["forces"],
+        calc_forces,
+        atol=1e-4,
```

Comment:
We've updated the tests to add more test cases, can you look at the model tests on main and copy the structure that's now there

Reply:
Updated